pkg/monitor/cluster/machineconfigpool_test.go

```diff
@@ -0,0 +1,100 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	mcofake "github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/fake"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitMachineConfigPoolUnmanagedNodes_TooManyNodes(t *testing.T) {
```

Comment:
Sorry for being nit monster here. Can you please write these tests in "loop" model. This is more pattern we follow and onverall separate functions for testcases is bit anti-pattern in go 
Something like this: 
```
for _, tt := range []struct {
		name                                     string
		mco                                       *[]mcv1.MachineConfigPool
		nodes                                     *[]corev1.Node
		expect                                    <someobject or metric>
	}{
	t.Run(tt.name, func(t *testing.T) {
			mco := mcofake.NewSimpleClientset(tt.mco)
                        nodes := fake.NewSimpleClientset(tt.nodes)

			controller := gomock.NewController(t)
			defer controller.Finish()

			... test logic
    })
}
```
```

Reply:
https://github.com/Azure/ARO-RP/pull/1450/commits/714d20458ddfe24fdee12ab5fa760621598145d0